File: library/src/main/java/zhan/library/delegate/ScrollHeaderDelegate.java
package zhan.library.delegate;

import android.view.View;



public class ScrollHeaderDelegate extends ScrollerDelegate {

  private static final int NORMAL_DURATION = 800;

  private View mTargetView;
  private boolean isScrollShow;
  private int mDuration = NORMAL_DURATION;

  public ScrollHeaderDelegate(View targetView) {
    super(targetView);
    mTargetView = targetView;
  }

  public void scrollShow() {
    if (!isScrollShow) {
      return;
    }
    int height = mTargetView.getMeasuredHeight();
    if(height <= 0) {
      return;
    }
    smoothScrollTo(0, height, 0, -height, mDuration);
    isScrollShow = false;
  }

  public void setScrollShow(boolean flag) {
    isScrollShow = flag;
  }

  public boolean isScrollShow() {
    return isScrollShow;
  }

  public void setDuration(int duration) {
    mDuration = duration;
  }
}

File: double-pull-delegate/src/main/java/zhan/pull/delegate/ScrollHeaderDelegate.kt
package zhan.pull.delegate

import android.view.View


class ScrollHeaderDelegate(private val mTargetView: View) : ScrollerDelegate(mTargetView) {
    var isScrollShow: Boolean = false
    private var mDuration = NORMAL_DURATION

    fun scrollShow() {
        if (!isScrollShow) {
            return
        }
        val height = mTargetView.measuredHeight
        if (height <= 0) {
            return
        }
        smoothScrollTo(0, height, 0, -height, mDuration)
        isScrollShow = false
    }

    fun setDuration(duration: Int) {
        mDuration = duration
    }

    companion object {

        private val NORMAL_DURATION = 800
    }
}
